### PR TITLE
Test for Record add_gene_panel

### PR DESCRIPTION
**Pull request by northwestwitch**
The other tests are already done

### CODE REVIEW

File: t/record_add_program_outfile_to_sample_info.t [hunk 1]
```diff
@@ -120,10 +120,54 @@ is( $sample_info{program}{$test_program_name}{path},
...
+);
+## Test
+is(
+    !exists $sample_info{sample}{$sample_id}{program}{$test_program_name}
+      {$infile},
+    1,
+    q{Sample level hash key not created when infile is not provided}
+);
+
+is(
```
Comment (henrikstranneheim) [inline t/record_add_program_outfile_to_sample_info.t]: You should be able to loop over this. Something like:
my %test_no_infile = (outdirectory => q{Value to sample level outdirectory not assigned});
'#' and so on for the parameters you want to test and then:

while (my ($parameter, $test_comment) = each %test_no_infile) {
my $test_result = $sample_info{sample}{$sample_id}{program}{$test_program_name}
+      {$infile}{$parameter};
is($test_result, undef, $test_comment);
}

**Comment by northwestwitch** [comment]
I've realized that while I can loop over the un-assigned params when you are not using an infile (because they all return undef), you can't do the same for when you have the infile (because you expect defined values, one for each cycle). So I left the last part of the code as it was.

File: t/record_add_program_outfile_to_sample_info.t [hunk 2]
```diff
@@ -120,10 +120,40 @@ is( $sample_info{program}{$test_program_name}{path},
...
+    }
+);
+## Test
+
+is( $sample_info{sample}{$sample_id}{program}{$test_program_name}{path},
+    $path, q{Assigned correct value to sample level path} );
+
+my %test_no_infile =
+  ( outdirectory => q{Value to sample level outdirectory not assigned} );
+$test_no_infile{outfile} .= q{Value to sample level outfile not assigned};
```
Comment (henrikstranneheim) [inline t/record_add_program_outfile_to_sample_info.t]: Continue with the key value assignment from the initial my instead. Like this:
my %test_no_infile = ( outdirectory => q{Value to sample level outdirectory not assigned},
                                     outfile => q{Value to sample level outfile not assigned},
                                     path => q{Value to sample level path not assigned},
                                      version => q{Value to sample level version not assigned},
                                   );

File: t/record_add_gene_panel.t [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+    {
+      aggregate_gene_panel_file => $aggregate_gene_panel_file,
+      aggregate_gene_panels_key => $aggregate_gene_panels_key,
+      family_id                 => $family_id_test,
+      program_name => $program_name_test,
+      sample_info_href => \%sample_info,
+    }
+);
+
+is( exists $sample_info{$program_name_test}{$aggregate_gene_panels_key}{gene_panel},
```
Comment (henrikstranneheim) [inline t/record_add_gene_panel.t]: You are only testing the existence of the key here and not the value

also add a test for the value:
is( exists $sample_info{$program_name_test}{$aggregate_gene_panels_key}{gene_panel}, q{gene_panel_test}, q{Gene panel added to $sample_info} );

Double check that the expected return actually is q{gene_panel_test}.

Comment (northwestwitch) [reply t/record_add_gene_panel.t]: It's actually not q{gene_panel_test}, it's a hash.

If I print this hash of hashes I get this:
first level,key1:TEST
	second level, key2:path,value2-->path_to_gene_panel.txt (shortened by me)
first level,key1:vcfparser
	second level, key2:TEST,value2-->HASH(0x7fbffacae400)

I'm not really understanding what I'm supposed to retrieve here..

**Comment by henrikstranneheim** [comment]
@northwestwitch Can you pull these changes to your local branch and see if they make pass.

**Comment by northwestwitch** [comment]
Yes, now I see. Didn't think of using the header info to provide the third level keys. I'll work on the other test now!

**Comment by henrikstranneheim** [comment]
Excellent!

**Pull request merged**